src/app/modules/verification/verification.controller.ts

```diff
@@ -412,7 +477,24 @@ export const _handleOtpVerification: ControllerHandler<
   return (
     FormService.retrieveFormById(formId)
       // Step 2: Verify the otp sent over by the client
-      .andThen(() => VerificationService.verifyOtp(transactionId, fieldId, otp))
+      .andThen(() => {
+        let getFieldFromTransactionFx
+        let incrementFieldRetriesFx
+        if (fieldId === PAYMENT_CONTACT_FIELD_ID) {
+          getFieldFromTransactionFx = getPaymentContactFieldFromTransaction
+          incrementFieldRetriesFx = incrementPaymentFieldRetriesWrapper
+        } else {
+          getFieldFromTransactionFx = getFieldFromTransaction
+          incrementFieldRetriesFx = incrementFormFieldRetriesWrapper
+        }
```

Comment:
I'm starting to think that maybe it's not necessary to abstract the functional conditioning away from the service layer into the controller layer. In any case, the check for whether `fieldId === PAYMENT_CONTACT_FIELD_ID` must occur at some point. In this case, we've complicated the service interfaces at the benefit of having a more uniform service layer function. 

On the contrary I feel that it would be better to have simpler controller functions, and have the service functions handle more of the detailed logic. This is also because we don't actually have separation of payment/form fields on the routes, and so we might expect that the model-level implementation detail of `FormField` vs `PaymentField` should be transparent to the controller. I'm thinking that the service layer should decide which method to call between the `FormField` and `PaymentField` variants. Better still, we don't need to modify any of the existing interfaces, since all the service functions take in `fieldId` anyway, so the check can just be shifted to service functions.  What do you think?

Reply:
Yea since now we're using the same endpoints for both normal form fields and payment contact field, then it makes sense for the logic to be left till the service level... 😢 okay i'll refactor and update the tests

Reply:
Done in [8b1eed0](https://github.com/opengovsg/FormSG/pull/6003/commits/8b1eed09b66bddbb99cdc30b40f25646680e70ab).